### PR TITLE
moved src/Tests to tests and use autoload-dev

### PR DESCRIPTION
**Pull request by ivoba**
| Q             | A
| ------------- | ---
| Bug fix?      | [no]
| New feature?  | [no]
| BC breaks?    | [no]
| Deprecations? | [no]
| Tests pass?   | [yes]
| Fixed tickets | [152]
| License       | MIT

Moved src/Tests to tests and use composers autoload-dev.  
Further i added a .gitattributes to exclude tests from dist archives.

### CODE REVIEW

**Comment by ivoba** [comment]
hmm appveyour fails, i dont know this service.  
any ideas?

**Comment by csarrazi** [comment]
It seems there's something weird with how PHPUnit is run on AppVeyor (CI for testing Windows builds).

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+# Ignore all test and documentation with "export-ignore".
+/.gitattributes     export-ignore
+/.gitignore         export-ignore
+/.travis.yml        export-ignore
+/.styleci.yml       export-ignore
+/appveyor.yml       export-ignore
+/phpunit.xml.dist   export-ignore
+/.scrutinizer.yml   export-ignore
+/tests              export-ignore
```
Comment (csarrazi) [inline .gitattributes]: Missing newline.

Comment (ivoba) [inline .gitattributes]: im sorry, what do you mean?
a newline at the end of the file?

**Comment by ivoba** [comment]
I would guess AppVeyor cant find the tests or the phpunit.xml.dist file. Strange.

**Comment by csarrazi** [comment]
What actually happens is that appveyor is actually using the github API to download the repository data. Thus, in the background, a git archive is actually run, in order to provide a repository snapshot, which means that the files are deleted.

See https://www.appveyor.com/docs/how-to/repository-shallow-clone#downloading-repository-via-github-or-bitbucket-api

Thus, the `phpunit.xml.dist` file is not found after the archive is downloaded.

**Comment by csarrazi** [comment]
I don't understand why the `appveyor.yml` file is still found (didn't check whether you did a typo for that file).

**Comment by ivoba** [comment]
Ok, i changed `shallow_clone: true` to `clone_depth: 1`.
Now it works. :)

**Comment by csarrazi** [comment]
Could you change the clone depth to 10? A too low value may actually break PR builds if there are too many commits in the PR.

It should be good to merge after that! :)

**Comment by ivoba** [comment]
Done

**Comment by csarrazi** [comment]
Thanks @ivoba !

**Comment by ivoba** [comment]
Will you tag a new release?

**Comment by csarrazi** [comment]
@ivoba, the next release for the `master` branch will be 2.1.0, which is basically waiting for #138, which is actually depending on guzzle-services upgrading to Guzzle 6.1 (On which I'm currently working).

I could change this, and tag a 2.1 release, and move the features for release 2.2, in a few weeks/months.

**Comment by ivoba** [comment]
yes, if it would take weeks/months then i would welcome an intermediate release.

as i see there were quite a few commits since the last release, so it would probably be worth it.

**Comment by csarrazi** [comment]
Done! :)

**Pull request merged**